bundles/org.openhab.binding.bluetooth.govee/src/main/java/org/openhab/binding/bluetooth/govee/internal/GoveeHygrometerHandler.java

```diff
@@ -0,0 +1,429 @@
+
+    private GoveeHygrometerConfiguration config = new GoveeHygrometerConfiguration();
+    private GoveeModel model = GoveeModel.H5074;// we use this as our default model
+
+    private CompletableFuture<?> initializeJob = CompletableFuture.completedFuture(null);// initially set to a dummy
+                                                                                         // future
+    private Future<?> scanJob = CompletableFuture.completedFuture(null);
+    private Future<?> keepAliveJob = CompletableFuture.completedFuture(null);
+
+    public GoveeHygrometerHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+        config = getConfigAs(GoveeHygrometerConfiguration.class);
+
+        Map<String, String> properties = thing.getProperties();
+        String modelProp = properties.get(Thing.PROPERTY_MODEL_ID);
+        model = GoveeModel.H5074;
+        if (modelProp != null) {
+            try {
+                model = GoveeModel.valueOf(modelProp);
+            } catch (IllegalArgumentException ex) {
+                // ignore
+            }
+        }
+
+        logger.debug("Initializing Govee Hygrometer {} model: {}", address, model);
+        initializeJob = RetryFuture.composeWithRetry(this::createInitSettingsJob, scheduler)//
+                .thenRun(() -> {
+                    updateStatus(ThingStatus.ONLINE);
+                });
+        scanJob = scheduler.scheduleWithFixedDelay(() -> {
+            try {
+                if (initializeJob.isDone() && !initializeJob.isCompletedExceptionally()) {
+                    logger.debug("refreshing temperature, humidity, and battery");
+                    refreshBattery().join();
+                    refreshTemperatureAndHumidity().join();
+                    connectionTaskExecutor.execute(device::disconnect);
+                    updateStatus(ThingStatus.ONLINE);
+                }
+            } catch (RuntimeException ex) {
+                logger.warn("unable to refresh", ex);
+            }
+        }, 0, config.refreshInterval, TimeUnit.SECONDS);
+        keepAliveJob = connectionTaskExecutor.scheduleWithFixedDelay(() -> {
+            if (device.getConnectionState() == ConnectionState.CONNECTED) {
+                try {
+                    GoveeMessage message = new GoveeMessage((byte) 0xAA, (byte) 1, null);
+                    writeCharacteristic(SERVICE_UUID, KEEP_ALIVE_CHAR_UUID, message.getPayload(), false);
+                } catch (RuntimeException ex) {
+                    logger.warn("unable to send keep alive", ex);
+                }
+            }
+        }, 1, 2, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void dispose() {
+        try {
+            throw new Exception();
+        } catch (Exception ex) {
+            logger.debug("Disposing Govee Hygrometer {} model: {}", address, model, ex);
+        }
```

Comment:
What's the purpose of this?

Reply:
wow, I totally forgot to remove that. It was strictly for debugging.

Reply:
done